game/src/main/kotlin/world/gregs/voidps/world/interact/entity/npc/combat/type/GiantMole.kts

```diff
@@ -0,0 +1,212 @@
+
+val acceptedTiles = listOf(
+    Tile(3005, 3376, 0),
+    Tile(2999, 3375, 0),
+    Tile(2996, 3377, 0),
+    Tile(2989, 3378, 0)
+)
+
+val giantMoleLair = areas["giant_mole_lair"]
+val gianMoleSpawns = areas["giant_mole_spawn_area"]
+val initialCaveTile: Tile = Tile(1752, 5237, 0)
+
+inventoryOption("Dig") {
+    val playerTile: Tile = player.tile
+    if(!acceptedTiles.contains(playerTile)) {
+        return@inventoryOption
+    }
+    player.setAnimation("dig_with_spade")
+    player.open("warning_dark")
+}
+
+objectOperate("Climb", "giant_mole_lair_escape_rope") {
+    player.setAnimation("climb_up")
+    player.tele(acceptedTiles.random())
+}
+
+interfaceOption(component = "proceed", id = "warning_dark") {
+    player.tele(initialCaveTile, clearInterfaces = true)
+}
+
+interfaceOption(component = "stayout", id = "warning_dark") {
+    player.closeInterfaces()
+}
+
+npcCombatHit("giant_mole") {
+    val currentHealth = it.levels.get(Skill.Constitution)
+    var shouldBurrow = false
+    if (it.fightStyle == "magic" && damage != 0) {
+        shouldBurrow = shouldBurrowAway(currentHealth)
+    } else if (it.fightStyle != "magic") {
+        shouldBurrow = shouldBurrowAway(currentHealth)
+    }
+    if (shouldBurrow && !it.hasClock("awaiting_mole_burrow_complete")) {
+        it.start("awaiting_mole_burrow_complete", 4)
+        giantMoleBurrow(it)
+    }
+}
+
+fun giantMoleBurrow(mole: NPC) {
+    for (attacker in mole.attackers) {
+        attacker.mode = EmptyMode
+    }
+    mole.attackers.clear()
+    var tileToDust = getTotalDirection(mole.facing, mole.tile)
+    World.queue("await_mole_to_face", 1) {
+        if (tileToDust == Tile.EMPTY) {
+            logger.info { "failed to get facing tile for Giant Mole, using default tile." }
+            tileToDust = initialCaveTile
+        }
+        mole.face(tileToDust)
+        World.queue("display_burrow_dust", 1) {
+            if (shouldThrowDirt()) {
+                handleDirtOnScreen(mole.tile)
+            }
+            mole.setAnimation("giant_mole_burrow")
+            areaSound("giant_mole_burrow_down", mole.tile)
+            areaGraphic("burrow_dust", tileToDust)
+            World.queue("await_mole_burrowing", 1) {
+                val newLocation = gianMoleSpawns.random(mole)
+                mole.tele(newLocation!!)
+                mole.setAnimation("mole_burrow_up")
+            }
+        }
+    }
+}
+
+fun getTotalDirection(facing: Direction, moleTile: Tile): Tile {
+    return when (facing) {
+        Direction.NORTH -> Tile(moleTile.x, moleTile.y + 1)
+        Direction.SOUTH -> Tile(moleTile.x, moleTile.y - 1)
+        Direction.EAST -> Tile(moleTile.x + 1, moleTile.y)
+        Direction.WEST -> Tile(moleTile.x - 1, moleTile.y)
+        Direction.NORTH_EAST, Direction.SOUTH_EAST -> Tile(moleTile.x + 1, moleTile.y)
+        Direction.NORTH_WEST, Direction.SOUTH_WEST -> Tile(moleTile.x - 1, moleTile.y)
+        Direction.NONE -> Tile.EMPTY
+    }
+}
```

Comment:
`moleTile.add(direction.delta)`

Reply:
I can remove this function now, because i will change the mole to get a random direction. so this isn't needed

Reply:
> `moleTile.add(direction.delta)`

Sorry may be a really simple question to ask, but could to explain what `direction.delta` does?

Reply:
Delta's are essentially the same as a Tile with an x, y, level coordinate but instead they represent differences between two tiles. (and technically the real difference is Delta's can have negative values, where Tile's cannot)

For example if you have a player at 100, 100 and an npc at 105, 106 then the delta from player -> npc would be 5, 6, and from npc -> player -5, -6

Same applies for directions, north is (0, 1), south (0, 1), east (1, 0), west (-1, 0) etc... so you can do `player.tile.add(Direction.NORTH_WEST.delta)` to get the tile north-west of the player (although in reality there is a helper so that player.tile.add(Direction.NORTH_WEST)` works without needing to type delta


so NORTH.delta is essentially Tile(0, 1), NORTH_WEST = Tile(-1, 1) etc... However due to technicalities Tile's can't have negative x/y's so Delta's exist as an alternative.

Delta's 
Delta's are also good for when doing differences between two characters, so you can do `player.tile.delta(npc.tile)` if player.tile is Tile(0, 0)

Reply:
Thank you for that, i understand now just wanted to make sure i knew what it did for future reference, and sorry for the simple question